packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -796,22 +833,56 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
       },
       roles: [cdk.Fn.ref('UserPoolClientRole')],
     });
+
     this.hostedUIProvidersCustomResourceLogPolicy.addDependsOn(this.hostedUIProvidersCustomResourcePolicy);
 
+    // iam policy for hostedUIProvider Lambda Function to get OAuth secrets
+
+    this.hostedUIProviderCustomResourceSecretsPolicy = new iam.CfnPolicy(this, 'hostedUIProvidersCustomResourceSecretPolicy', {
+      policyName: cdk.Fn.join('-', [cdk.Fn.ref('UserPool'), 'hostedUIProvidersCustomResourceSecretPolicy']),
+      policyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Effect: 'Allow',
+            Action: ['ssm:GetParameter'],
+            Resource: cdk.Fn.join('', [
+              'arn:aws:ssm:',
+              cdk.Fn.ref('AWS::Region'),
+              ':',
+              cdk.Fn.ref('AWS::AccountId'),
+              ':parameter',
+              cdk.Fn.sub(path.posix.join('/amplify', '${appId}', '${env}', 'AMPLIFY_${resourceName}_'), {
+                appId: cdk.Fn.ref(`${oAuthSecretsPathAmplifyAppIdKey}`),
+                env: cdk.Fn.ref('env'), // this is dependent on the Amplify env name being a parameter to the CFN template which should always be the case
+                resourceName: cdk.Fn.ref('resourceName'),
+              }),
+              `${oAuthObjSecretKey}`,
```

Comment:
would prefer to define this whole thing using just fn.sub instead of a combo of fn.join and fn.sub